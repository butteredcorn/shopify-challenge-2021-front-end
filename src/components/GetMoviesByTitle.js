import { useEffect } from 'react'
import '../styles/GetMoviesByTitle.css'
import styled from 'styled-components'

const MoviesContainer = styled.div`
    background-color: ${props => props.theme.bg.secondary};
    color: ${props => props.theme.text.tertiary};
    border: 1px solid ${props => props.theme.bg.primary};
`

const GetMoviesByTitle = ({keyword, loading, error, data, nominate, nominations}) => {

    useEffect(() => {
        console.log(data)
    })

    if (loading) return null;
    if (error) return `${error}.`;

    return (
        <MoviesContainer className="container">
            {data && data.length > 0 && <div>
                <h3 className="heading">Results for "{keyword}"</h3>
                <ul className="movies">
                {data.map((movie) => 
                    <li className="movie" key={movie.imdbID}><span>{movie.Title}</span> <span>({movie.Year})</span> <span><button onClick={() => nominate([...nominations, movie])}>Nominate</button></span> <span><button>View Details</button></span></li>
                )}
                </ul>
            </div>}
            {data && data.length == 0 && <div>
                <h3 className="heading">Results</h3>
                <p>Movie results will show here after you search.</p>
            </div>}
        </MoviesContainer>
    )
}

export default GetMoviesByTitle;