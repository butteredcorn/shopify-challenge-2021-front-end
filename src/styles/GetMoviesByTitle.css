.container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 1rem;
}

.heading {
    text-align: left;
}

.movies {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}

.movie { 
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
}

.movie > * {
    padding-right: 0.5rem;
}