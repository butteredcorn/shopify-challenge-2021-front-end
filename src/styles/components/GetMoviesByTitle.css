.movies-container {
    display: flex;
    flex-direction: column;
    padding: 1rem;
}

.heading {
    text-align: left;
}

.movies {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}

.movie { 
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
}

.movie > * {
    padding-right: 0.5rem;
}

@media only screen and (max-width: 600px) {

    .movies-container {
        margin-top: 0;
        margin-bottom: 0;
    }

    .movies {
        list-style: none;
    }

    .movie > * {
        padding-right: 0;
    }
}