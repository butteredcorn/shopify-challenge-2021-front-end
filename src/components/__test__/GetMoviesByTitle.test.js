import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import ThemesProvider from '../../providers/ThemeProvider'
import GetMoviesByTitle from '../GetMoviesByTitle';

afterEach(cleanup);

it('renders GetMoviesByTitle component with "Movie results will show here after you search." if prop "data" is set to empty array.', () => {
  render(<ThemesProvider><GetMoviesByTitle data={[]} /></ThemesProvider>);
  const message = screen.getByText(/Movie results will show here after you search./i);
  expect(message).toBeInTheDocument();
});

it('renders GetMoviesByTitle component without "Movie results will show here after you search." if prop "data" is set to an array with two movie objects.', () => {
    render(<BrowserRouter><ThemesProvider><GetMoviesByTitle data={[{Title: "Shopify Challenge - Front End Developer!", Year: "2021", imdbID: "the_shoppies"}, {Title: "Hello World!", Year: "2021", imdbID: "someID"}]} nominations={[{Title: "Shopify Challenge - Front End Developer!", Year: "2021", imdbID: "the_shoppies"}]}/></ThemesProvider></BrowserRouter>);
    const nonExistantMessage = screen.queryByText(/Movie results will show here after you search./i);
    expect(nonExistantMessage).toBeNull();
});

it('expect GetMoviesByTitle to render an error if an error is passed in as a prop', () => {
    render(<ThemesProvider><GetMoviesByTitle error={new Error("Hello World!")} data={[{Title: "Shopify Challenge - Front End Developer!", Year: "2021", imdbID: "the_shoppies"}, {Title: "Hello World!", Year: "2021", imdbID: "someID"}]} nominations={[{Title: "Shopify Challenge - Front End Developer!", Year: "2021", imdbID: "the_shoppies"}]}/></ThemesProvider>);
    const message = screen.getByText(/Hello World!/i);
    expect(message).toBeInTheDocument();
});

it('matches a loading snapshot', () => {
    const json = renderer.create(<ThemesProvider><GetMoviesByTitle loading={true} data={[{Title: "Shopify Challenge - Front End Developer!", Year: "2021", imdbID: "the_shoppies"}, {Title: "Hello World!", Year: "2021", imdbID: "someID"}]} /></ThemesProvider>);
    expect(json).toMatchSnapshot();
});

it('matches a snapshot', () => {
  const json = renderer.create(<BrowserRouter><ThemesProvider><GetMoviesByTitle data={[{Title: "Shopify Challenge - Front End Developer!", Year: "2021", imdbID: "the_shoppies"}, {Title: "Hello World!", Year: "2021", imdbID: "someID"}]} nominations={[{Title: "Shopify Challenge - Front End Developer!", Year: "2021", imdbID: "the_shoppies"}]}/></ThemesProvider></BrowserRouter>).toJSON();
  expect(json).toMatchSnapshot();
});