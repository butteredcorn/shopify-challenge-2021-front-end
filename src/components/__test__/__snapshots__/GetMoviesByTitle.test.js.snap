// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a loading snapshot 1`] = `
<div
  className="sc-gsDJrp dbPJRZ movies-container"
>
  <p>
    Loading ...
  </p>
</div>
`;

exports[`matches a snapshot 1`] = `
<div
  className="sc-gsDJrp dbPJRZ movies-container"
>
  <div
    className="movies-container-inner"
  >
    <h3
      className="sc-dkPtyc iJlIGG heading"
    >
      2
       Results for "
      "
    </h3>
    <ul
      className="movies"
    >
      <li
        className="movie"
      >
        <a
          href="/movie"
          onClick={[Function]}
        >
          Shopify Challenge - Front End Developer!
        </a>
         
        <span>
          (
          2021
          )
        </span>
         
        <span />
      </li>
      <li
        className="movie"
      >
        <a
          href="/movie"
          onClick={[Function]}
        >
          Hello World!
        </a>
         
        <span>
          (
          2021
          )
        </span>
         
        <span>
          <button
            onClick={[Function]}
          >
            Nominate
          </button>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
